frontend/src/app/components/address-labels/address-labels.component.ts

```diff
@@ -98,44 +98,52 @@ export class AddressLabelsComponent implements OnChanges {
   }
 
   detectMultisig(script: string) {
-    if (!script) {
-      return;
-    }
-    const ops = script.split(' ');
-    if (ops.length < 3 || ops.pop() !== 'OP_CHECKMULTISIG') {
-      return;
-    }
-    const opN = ops.pop();
-    if (!opN.startsWith('OP_PUSHNUM_')) {
-      return;
-    }
-    const n = parseInt(opN.match(/[0-9]+/)[0], 10);
-    if (ops.length < n * 2 + 1) {
-      return;
-    }
-    // pop n public keys
-    for (let i = 0; i < n; i++) {
-      if (!/^0((2|3)\w{64}|4\w{128})$/.test(ops.pop())) {
-        return;
-      }
-      if (!/^OP_PUSHBYTES_(33|65)$/.test(ops.pop())) {
-        return;
-      }
+    const ms = parseMultisigScript(script);
+
+    if (ms) {
+      this.label = $localize`:@@address-label.multisig:Multisig ${ms.m}:multisigM: of ${ms.n}:multisigN:`;
     }
-    const opM = ops.pop();
-    if (!opM.startsWith('OP_PUSHNUM_')) {
+  }
+
+  handleVout() {
+    this.detectMultisig(this.vout.scriptpubkey_asm);
+  }
+}
+
+export function parseMultisigScript(script: string): void | { m: number, n: number } {
```

Comment:
Can you move this method into the `bitcoin.utils.ts` file instead.

Reply:
ok yeah, also looks cleaner not importing from a component